torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -174,7 +175,7 @@ std::shared_ptr<FutureMessage> ProcessGroupAgent::send(
 
   auto requestId = nextId();
   auto future = std::make_shared<FutureMessage>();
-  if (message.isRequest()) {
+  if (message.requiresResponse()) {
```

Comment:
how does it work when isRequest() is superset of requiresResponse() while  isRequest() is inside if(requiresResponse())?

Reply:
Sorry, didn't follow. Where is the inner `isRequest`?